gcovr/templates/source_page.html

```diff
@@ -16,6 +16,31 @@
       <th scope="row">Date:</th>
       <td>{{info.date}}</td>
     </tr>
+    {% if function_names %}
+    <tr>
+      <th scope="row">Goto function:</th>
+      <td>
+        <form name="gotoFunctionName">
+          <select name="selection">
+            <option selected="selected" value=""></option>
+        {% for function_name in function_names %}
+            <option value="#{{ function_name.1 }}">{{ function_name.0 }}, line {{ function_name.1 }}</a></option>
+        {% endfor %}
+          </select>
+        </form>
+        <script>
+          function gotoFunctionName () {
+            var menu = document.gotoFunctionName.selection;
+            if (menu.options[menu.selectedIndex].value != "") {
+              document.location.href = menu.options[menu.selectedIndex].value;
+              menu.selectedIndex = 0;
+            }
+          }
+          document.gotoFunctionName.selection.addEventListener('change', gotoFunctionName);
+        </script>
```

Comment:
We can't always use inline scripts, just like we can't always use inline CSS: the reports might be served from a HTTP server with an unsuitable Content Security Policy (CSP).

But I don't think we need any JavaScript – a list of links should be sufficient. The per-line branch coverage data uses a `<details>` element to create a popup without JS, maybe we can repurpose that here to save space.

So I think we could maybe replace this navigation tool by something like:

```html
<details>
  <summary>go to function</summary>
  <ul>
  {% for function_name in function_names %}
    <li><a href="#{{ function_name.1 }}">{{ function_name.0 }}</a></li>
  {% endfor %}
  </ul>
</details>
```

Reply:
I'll check this.

Reply:
Done.